src/oidc/oidc.util.ts

```diff
@@ -0,0 +1,190 @@
+      enabled: true,
+      // this is default function, change it
+      // allowedPolicy(ctx, client, token) {
+      //   if (client.clientAuthMethod === 'none' && token.clientId !== ctx.oidc.client.clientId) {
+      //     return false;
+      //   }
+      //   return true;
+      // },
+    },
+    /* resourceIndicators: {
+       enabled: true,
+        getResourceServerInfo(ctx, resourceIndicator) {
+        if (resourceIndicator === 'urn:api') {
+           return {
+             scope: 'read',
+             audience: 'urn:api',
+             accessTokenTTL: 1 * 60 * 60, // 1 hour
+             accessTokenFormat: 'jwt',
+           };
+         }
+
+         throw new Error('Invalid target');
+       },
+     }, */
+    /* To turn off default pages and routes */
+    devInteractions: {
+      enabled: false,
+    },
+  },
+  interactions: {
+    url(_, interaction) {
+      return `/interaction/${interaction.uid}`;
+    },
+  },
+  clientBasedCORS(ctx, origin, client) {
+    return false;
+  },
+
+  //   claims:{
+  //     profile: ['username','gender','birthdate','email'],
+  //   },
+
+  clients: [
+    {
+      client_id: process.env.CLIENT_ID,
+      client_secret: process.env.CLIENT_SECRET,
+      redirect_uris: [
+        `${process.env.HOST_NAME}:${process.env.HOST_PORT}/${process.env.OIDC_CALLBACK_ROUTE}`,
+      ],
+      // + other client properties
+      grant_types: [
+        'authorization_code',
+        'client_credentials',
+        'refresh_token',
+      ],
+      //   redirect_uris: [],
+      response_types: ['code'],
+    },
+  ],
+  pkce: {
+    methods: ['S256'],
+    required: () => false,
+  },
+  scopes: ['openid', 'offline_access', 'profile', 'email', 'phone', 'address'],
+
+  /*   findAccount: async function findAccount(ctx,sub,token){
+
+  //   }
+
+  //   interactions: {
+  //     url(ctx, interaction) {
+  //       // eslint-disable-line no-unused-vars
+  //       return `/interaction/${interaction.uid}`;
+  //     },
+  //   }, */
+
+  // own cookies.key needed rather than default
+  cookies: {
+    keys: [
+      'some secret key',
+      'and also the old rotated away some time ago',
+      'and one more',
+    ],
+  },
+
+  /*   claims: {
+  //     address: ['address'],
+  //     email: ['email', 'email_verified'],
+  //     phone: ['phone_number', 'phone_number_verified'],
+  //     profile: [
+  //       'birthdate',
+  //       'family_name',
+  //       'gender',
+  //       'given_name',
+  //       'locale',
+  //       'middle_name',
+  //       'name',
+  //       'nickname',
+  //       'picture',
+  //       'preferred_username',
+  //       'profile',
+  //       'updated_at',
+  //       'website',
+  //       'zoneinfo',
+  //     ],
+  //   },
+  //   features: {
+  //     devInteractions: { enabled: false }, // defaults to true
+
+  //     deviceFlow: { enabled: true }, // defaults to false
+  //     revocation: { enabled: true }, // defaults to false
+  //   }, */
```

Comment:
remove unless this is for document purposes, if this is for document purposes, change this to use JSDoc format.